test/gurka/test_search_filter.cc

```diff
@@ -182,3 +185,840 @@ TEST_F(SearchFilter, ExcludeRamp) {
+        (boost::format(
+             R"({"locations":[{"lat":%s,"lon":%s,"heading":90},{"lat":%s,"lon":%s}],"costing":"%s", "costing_options": {"%s": {"speed_types":["freeflow","constrained","predicted","current"]}}, "date_time":{"type":"%s", "value": "current"}})") %
+         std::to_string(closure_map.nodes.at("1").lat()) %
+         std::to_string(closure_map.nodes.at("1").lng()) %
+         std::to_string(closure_map.nodes.at("E").lat()) %
+         std::to_string(closure_map.nodes.at("E").lng()) % costing % costing % date_type)
+            .str();
+    auto result = gurka::route(closure_map, req_with_heading, reader);
+    gurka::assert::osrm::expect_steps(result, {"CD"});
+    gurka::assert::raw::expect_path(result, {"CD", "DE"});
+  }
+
+  // Close consecutive edges CD & DE
+  {
+    LiveTrafficCustomize close_edge = [](baldr::GraphReader& reader, baldr::TrafficTile& tile,
+                                         int index, baldr::TrafficSpeed* current) -> void {
+      close_single_edge(reader, tile, index, current, "CD", closure_map);
+      close_single_edge(reader, tile, index, current, "DE", closure_map);
+    };
+    test::customize_live_traffic_data(closure_map.config, close_edge);
+
+    const std::string& req_with_heading =
+        (boost::format(
+             R"({"locations":[{"lat":%s,"lon":%s,"heading":90},{"lat":%s,"lon":%s}],"costing":"%s", "costing_options": {"%s": {"speed_types":["freeflow","constrained","predicted","current"]}}, "date_time":{"type":"%s", "value": "current"}})") %
+         std::to_string(closure_map.nodes.at("1").lat()) %
+         std::to_string(closure_map.nodes.at("1").lng()) %
+         std::to_string(closure_map.nodes.at("E").lat()) %
+         std::to_string(closure_map.nodes.at("E").lng()) % costing % costing % date_type)
+            .str();
+    auto result = gurka::route(closure_map, req_with_heading, reader);
+    gurka::assert::osrm::expect_steps(result, {"AC", "AB", "BE"});
+    gurka::assert::raw::expect_path(result, {"AC", "AB", "BE"});
+
+    const std::string& req_disable_exclude_closures =
+        (boost::format(
+             R"({"locations":[{"lat":%s,"lon":%s,"heading":90,"search_filter":{"exclude_closures":false}},{"lat":%s,"lon":%s,"search_filter":{"exclude_closures":false}}],"costing":"%s", "costing_options": {"%s": {"speed_types":["freeflow","constrained","predicted","current"]}}, "date_time":{"type":"%s", "value": "current"}})") %
+         std::to_string(closure_map.nodes.at("1").lat()) %
+         std::to_string(closure_map.nodes.at("1").lng()) %
+         std::to_string(closure_map.nodes.at("E").lat()) %
+         std::to_string(closure_map.nodes.at("E").lng()) % costing % costing % date_type)
+            .str();
+    result = gurka::route(closure_map, req_disable_exclude_closures, reader);
+    gurka::assert::osrm::expect_steps(result, {"CD", "AC", "AB", "BE"});
+    gurka::assert::raw::expect_path(result, {"CD", "CD", "AC", "AB", "BE"});
```

Comment:
This test currently fails if expecting a uturn on `CD` (the actual path returned is `"CD", "AC", "AB", "BE"`) even with a heading of 90º (going towards `D`).

Seems like the uturn wouldn't occur with left-side driving?